pkg/container/vector/vector.go

```diff
@@ -72,1608 +65,757 @@ func (v *Vector) GetIsBin() bool {
-			tgt := v.getRawValueAt(next + int64(idx))
-			copy(tgt, src)
-		}
-	}
-
-	if hasNull && w.Nsp != nil {
-		for i := range sels {
-			if nulls.Contains(w.Nsp, uint64(sels[i])) {
-				nulls.Add(v.Nsp, uint64(oldLen+i))
-			}
-		}
-	}
-
-	return
-}
-
-// XXX Old UnionBatch is FUBAR.
-func UnionBatch(v, w *Vector, offset int64, cnt int, flags []uint8, m *mpool.MPool) (err error) {
-	if v.original {
-		return moerr.NewInternalErrorNoCtx("UnionBatch cannot be performed on orig vector")
-	}
-
-	curIdx := v.Length()
-	oldLen := uint64(curIdx)
-
-	if err = v.extend(cnt, m); err != nil {
-		return err
-	}
-
-	if v.GetType().IsTuple() {
-		vs := v.Col.([][]interface{})
-		ws := w.Col.([][]interface{})
-		for i, flag := range flags {
-			if flag > 0 {
-				vs = append(vs, ws[int(offset)+i])
-			}
-		}
-		v.Col = vs
-	} else if v.GetType().IsVarlen() {
-		tgt := MustTCols[types.Varlena](v)
-		for idx, flg := range flags {
-			if flg > 0 {
-				bs := w.GetBytes(offset + int64(idx))
-				if v.GetType().Width != 0 && len(bs) > int(v.GetType().Width) {
-					return moerr.NewOutOfRangeNoCtx("varchar/char ", "%v oversize of %v ", string(bs), v.GetType().Width)
-				}
-
-				if v.GetType().Width == 0 && (v.GetType().Oid == types.T_varchar || v.GetType().Oid == types.T_char) {
-					if len(bs) > 0 {
-						return moerr.NewOutOfRangeNoCtx("varchar/char ", "%v oversize of %v ", string(bs), 0)
-					}
-				}
-				tgt[curIdx], v.area, err = types.BuildVarlena(bs, v.area, m)
-				curIdx += 1
-			}
-		}
-	} else {
-		for idx, flg := range flags {
-			if flg > 0 {
-				src := w.getRawValueAt(offset + int64(idx))
-				tgt := v.getRawValueAt(int64(curIdx))
-				copy(tgt, src)
-				curIdx += 1
-			}
-		}
-	}
-
-	if w.Nsp != nil {
-		for idx, flg := range flags {
-			if flg > 0 {
-				if nulls.Contains(w.Nsp, uint64(offset)+uint64(idx)) {
-					nulls.Add(v.Nsp, oldLen)
-				}
-				// Advance oldLen regardless if it is null
-				oldLen += 1
-			}
-		}
-	}
-
-	return
-}
-
-// XXX Old Reset is FUBAR FUBAR.   I will put the code here just for fun.
-func Reset(v *Vector) {
-	/*
-		switch v.Typ.Oid {
-		case types.T_char, types.T_varchar, types.T_json, types.T_blob:
-			v.Col.(*types.Bytes).Reset()
-		default:
-			// WTF is going on?
-			*(*int)(unsafe.Pointer(uintptr((*(*emptyIntervade)(unsafe.Pointer(&v.Col))).word) + uintptr(strconv.IntSize>>3))) = 0
-		}
-	*/
-
-	// XXX Reset does not do mem accounting?
-	// I have no idea what is the purpose of Reset, so let me just Free it.
-	// Maybe Reset want to keep v.data and v.area to save an allocation.
-	// Let me do that ...
-	v.setupColFromData(0, 0)
-	v.area = v.area[:0]
-	// XXX What about Nsp? Original code does not do anything to Nsp, which seems OK assuming
-	// that will be set when we add data and we only test null within range of len(v.Col)
-	// but who knows ...
-}
-
-// XXX What are these stuff, who use it?
-func VecToString[T types.FixedSizeT](v *Vector) string {
+func vecToString[T types.FixedSizeT](v *Vector) string {
```

Comment:
who use this thing? :)

Reply:
Just so then fmt.Printf("%s\n", vec) shows something not so ugly. After all, there are always people who debug this output directly. So I added such a thing.